### PR TITLE
IT-2926: Create /repo and export NFS RW for gen3 butler summit services

### PR DESCRIPTION
**Pull request by hreinking**

### CODE REVIEW

File: hieradata/node/atarchiver.cp.lsst.org.yaml [hunk 1]
```diff
@@ -76,6 +76,25 @@ nfs::nfs_exports_global:
       139.229.160.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.165.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.175.128/25(ro,nohide,insecure,no_subtree_check,async,root_squash)
+  /data/repo:
+    clients: >-
+      %{facts.networking.ip}/32(ro,nohide,insecure,no_subtree_check,async,root_squash)
```
Comment (ktlim) [inline hieradata/node/atarchiver.cp.lsst.org.yaml:81]: This is for the self-mount?  Can it be rw instead, then?

File: hieradata/node/comcam-arctl01.cp.lsst.org.yaml [hunk 2]
```diff
@@ -99,13 +99,36 @@ nfs::nfs_exports_global:
       139.229.160.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.165.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.175.0/26(ro,nohide,insecure,no_subtree_check,async,root_squash)
+  /data/repo:
+    clients: >-
+      %{facts.networking.ip}/32(ro,nohide,insecure,no_subtree_check,async,root_squash)
```
Comment (ktlim) [inline hieradata/node/comcam-arctl01.cp.lsst.org.yaml:104]: And similarly here.

File: hieradata/node/comcam-arctl01.cp.lsst.org.yaml [hunk 3]
```diff
@@ -99,13 +99,31 @@ nfs::nfs_exports_global:
...
+      139.229.160.0/24(rw,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.165.0/24(rw,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.175.0/26(rw,nohide,insecure,no_subtree_check,async,root_squash)
+
+files:
+  /data/repo:
+    ensure: "directory"
+    mode: "0755"
+    owner: 73006 # saluser
+    group: 0
```
Comment (srp3rd) [inline hieradata/node/comcam-arctl01.cp.lsst.org.yaml:115]: Group should be 73006 #saluser

File: hieradata/node/atarchiver.cp.lsst.org.yaml [hunk 4]
```diff
@@ -76,6 +76,20 @@ nfs::nfs_exports_global:
...
+      139.229.160.0/24(rw,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.165.0/24(rw,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.175.0/26(rw,nohide,insecure,no_subtree_check,async,root_squash)
+
+files:
+  /data/repo:
+    ensure: "directory"
+    mode: "0755"
+    owner: 73006 # saluser
+    group: 0
```
Comment (srp3rd) [inline hieradata/node/atarchiver.cp.lsst.org.yaml:92]: group should be 73006 #saluser

File: hieradata/node/atarchiver.cp.lsst.org.yaml [hunk 5]
```diff
@@ -76,6 +76,25 @@ nfs::nfs_exports_global:
       139.229.160.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.165.0/24(ro,nohide,insecure,no_subtree_check,async,root_squash)
       139.229.175.128/25(ro,nohide,insecure,no_subtree_check,async,root_squash)
+  /data/repo:
+    clients: >-
+      %{facts.networking.ip}/32(ro,nohide,insecure,no_subtree_check,async,root_squash)
+      139.229.146.0/24(rw,nohide,insecure,no_subtree_check,async,root_squash)
```
Comment (ktlim) [inline hieradata/node/atarchiver.cp.lsst.org.yaml:82]: Will these mandate NFS v4?  That is necessary for locking (or else we have to ensure a bunch of other things are correct for NFS v3).

**Review by ktlim — APPROVED** [review]

**Review by srp3rd — CHANGES_REQUESTED** [review]

File: hieradata/node/comcam-arctl01.cp.lsst.org.yaml (hunk 3 above)
Comment (hreinking) [reply hieradata/node/comcam-arctl01.cp.lsst.org.yaml]: change applied

File: hieradata/node/atarchiver.cp.lsst.org.yaml (hunk 4 above)
Comment (hreinking) [reply hieradata/node/atarchiver.cp.lsst.org.yaml]: change applied

File: hieradata/node/atarchiver.cp.lsst.org.yaml (hunk 1 above)
Comment (hreinking) [reply hieradata/node/atarchiver.cp.lsst.org.yaml]: change applied

File: hieradata/node/comcam-arctl01.cp.lsst.org.yaml (hunk 2 above)
Comment (hreinking) [reply hieradata/node/comcam-arctl01.cp.lsst.org.yaml]: change applied

File: hieradata/node/atarchiver.cp.lsst.org.yaml (hunk 5 above)
Comment (hreinking) [reply hieradata/node/atarchiver.cp.lsst.org.yaml:82]: I believe it does, the puppet package at least is for nfs4

**Review by srp3rd — APPROVED** [review]

**Pull request merged**